zilliqa/src/consensus.rs

```diff
@@ -625,6 +629,27 @@ impl Consensus {
 
             self.save_highest_view(block.hash(), block.number(), proposal_view)?;
 
+            if let Some(buffered_votes) = self.buffered_votes.remove(&block.hash()) {
```

Comment:
It seems unfortunate that this edge case is injected here, where processing a proposal can respond with another proposal - it might be worth thinking of a way to make sure the system will just 'settle' without having to buffer neccessarily

Reply:
I think it makes sense for a proposal to result in another proposal if you accept that the node might be buffering sufficient votes for the first proposal to immediately form a supermajority.

I guess another way of representing this could be to buffer the votes in `Node`, rather than `Consensus`, then the return type of `proposal` wouldn't change. I'm not sure that would be much clearer though?